### ISSUE TITLE
Missing API KEY input in settings

### ISSUE THREAD

**Issue by snwfdhmp** [reworkd/AgentGPT#583]
### Please check that this issue hasn't been reported before.

- [X] I searched previous [Bug Reports](https://github.com/reworkd/AgentGPT/labels/bug) didn't find any similar reports.

### Expected Behavior

Settings should have an API KEY input

### Current behaviour

Settings doesn't have an API KEY input

<img width="572" alt="image" src="https://github.com/reworkd/AgentGPT/assets/23099246/c0d1a883-9c9e-49b4-bee1-46008eabed33">

### Steps to reproduce

Click on settings on the bottom left

### Which Operating Systems are you using?

- [ ] Android
- [ ] iPhone/iPad
- [ ] Linux
- [X] macOS
- [ ] Windows

### Acknowledgements

- [X] My issue title is concise, descriptive, and in title casing.
- [X] I have searched the existing issues to make sure this bug has not been reported yet.
- [X] I am using the latest version of AgentGPT.
- [X] I have provided enough information for the maintainers to reproduce and diagnose the issue.

**Comment by MentalGear** [comment]
I also found this very disappointing. Was it removed on purpose to make people sign-up?
I really hope they bring it back for the segment of people who enjoy using AgentGPT privately with their own key and in their own browser without outside analytics, which I think one of the original goals was.

**Comment by MentalGear** [comment]
Commit https://github.com/reworkd/AgentGPT/commit/bfed4390f1157f708496a7fe118cfdc5fa756cd5 states that the option for custom API keys have been removed because of "issues" but does not details what issues these are. Could you elaborate @asim-shrestha so maybe we can help with this?

**Comment by valimikayilov** [comment]
> Commit [bfed439](https://github.com/reworkd/AgentGPT/commit/bfed4390f1157f708496a7fe118cfdc5fa756cd5) states that the option for custom API keys have been removed because of "issues" but does not detail what issues these are. Could you elaborate @asim-shrestha so maybe we can help with this?

I guess the agentGPT is becoming for profit project: 
<img width="1333" alt="image" src="https://github.com/reworkd/AgentGPT/assets/24419936/6f4a4ffa-60c0-4599-865b-94df791d3c15">

**Comment by RogerThiede** [comment]
I suspect there are not many people who are sponsoring this GitHub project, and making the publicly hosted app require subscriptions would be a viable avenue for maintaining this project.
For developers, since this is an open source project, you can fork it and host it yourself by setting the server side OPENAI_API_KEY.

**Comment by MentalGear** [comment]
Indeed, I can understand the decision to have a commercial avenue, since open source often doesn't cover time spend. Would have been great though to keep the openAPI key setting in a hidden/hot key setting or so.

Especially since local setup seems not that straightforward: Trying to get it to run using by following the readme and run ./setup.sh --local, but I'm getting the following error:

<img width="1114" alt="Screenshot 2023-05-25 at 11 53 02" src="https://github.com/reworkd/AgentGPT/assets/2837147/58f5beab-3908-44cd-bb72-94711ceb3365">

**Comment by floriank** [comment]
This is indeed a bit disappointing for people who sponsor the project here on github.

**Comment by asim-shrestha** [comment]
Hey folks, happy to shed some light here. This was of course a decision we spent quite a bit of time with

Theres 2 dimensions to this decision:

---

1: To answer @MentalGear , the "issues" were in regards to the sheer number of OpenAI errors we were receiving daily: 
<img width="1222" alt="image" src="https://github.com/reworkd/AgentGPT/assets/50181239/544cd1f0-0987-4210-af09-472456ba36bb">

The above is our sentry metrics for errors related to but not limited to
- Free tier API key limit exhaustion
- Free tier API key rate limiting
- Users missing GPT-4 access 

Although we've tried to make the above problems evident whenever this occurred, ultimately our support lines became flooded with the same API key related questions. One of our mantras has always been customer accessibility and frictionless use and we ultimately felt that providing this option was harming this. Note these issues became further amplified when we moved to our Python backend from running OpenAI client side (All free tier use was running through the same server so all free tier API keys were immediately rate limited)

---

2: Costs. We owe a lot to our sponsors such as @floriank for helping keep us afloat. Unfortunately, sponsors alone aren't enough. At our peak the OpenAI bill was over 1k USD a day which was just not sustainable. This is why we've had to introduce features like loop limits, run limits, token limits, etc (Which we'll be removing from the open source repo). We want to support more models and more features in the future, and unfortunately this just won't be sustainable without a subscription model.

---

With that being said though, the open source community is incredibly important to us. AgentGPT will always be free and open to run locally. Unfortunately this has gotten a bit more difficult after our Python migration but we'll create something to help mitigate this within the next two days. Additionally, me and @awtkns would be happy to sit down and chat with anyone who has opinions on how we can mitigate the above issues to keep this feature alive. Please don't hesitate to reach out to me at asim@reworkd.ai or adam at adam@reworkd.ai directly. Would love to hear your thoughts @floriank @MentalGear

**Comment by sev7enITA** [comment]
Hello,
just found the same...  it least a news to inform about.

**Comment by harishruparel** [comment]
Hi,
Would love to know if this has changed. Still no option to add my OpenAI APIKey to AgentGPT. Look forward to know if there is any alternate method. 


![Image](https://github.com/reworkd/AgentGPT/assets/25629398/268aa1e7-e292-4d1d-8395-306ca95a1314)

**Comment by valimikayilov** [comment]
> Hi, Would love to know if this has changed. Still no option to add my OpenAI APIKey to AgentGPT. Look forward to know if there is any alternate method.
> 
> ![Image](https://user-images.githubusercontent.com/25629398/243654527-268aa1e7-e292-4d1d-8395-306ca95a1314.png)

you just need to download the repository to your local machine and launch it locally. I have a video on youtube for that:  https://www.youtube.com/watch?v=vf-51Dbr8Jg

However, there are a lot of changes to the AgentGPT since that video is made. So, the updated version will come soon and it will be filmed better.

**Comment by valimikayilov** [comment]
> Indeed, I can understand the decision to have a commercial avenue, since open source often doesn't cover time spend. Would have been great though to keep the openAPI key setting in a hidden/hot key setting or so.
> 
> Especially since local setup seems not that straightforward: Trying to get it to run using by following the readme and run ./setup.sh --local, but I'm getting the following error:
> 
> <img alt="Screenshot 2023-05-25 at 11 53 02" width="1114" src="https://user-images.githubusercontent.com/2837147/240875322-58f5beab-3908-44cd-bb72-94711ceb3365.png">

@MentalGear can you copy-paste the terminal?

**Comment by MentalGear** [comment]
> > Hi, Would love to know if this has changed. Still no option to add my OpenAI APIKey to AgentGPT. Look forward to know if there is any alternate method.
> > ![Image](https://user-images.githubusercontent.com/25629398/243654527-268aa1e7-e292-4d1d-8395-306ca95a1314.png)
> 
> Hi, you just need to download the repository to your local machine and launch it locally. I have a video on youtube for that: https://www.youtube.com/watch?v=vf-51Dbr8Jg
> 
> However, there are a lot of changes to the AgentGPT since that video is made. So, the updated version will come soon and it will be filmed better.

Yes, an updated video would be much appreciated!

**Comment by valimikayilov** [comment]
> > > Hi, Would love to know if this has changed. Still no option to add my OpenAI APIKey to AgentGPT. Look forward to know if there is any alternate method.
> > > ![Image](https://user-images.githubusercontent.com/25629398/243654527-268aa1e7-e292-4d1d-8395-306ca95a1314.png)
> > 
> > 
> > Hi, you just need to download the repository to your local machine and launch it locally. I have a video on youtube for that: https://www.youtube.com/watch?v=vf-51Dbr8Jg
> > However, there are a lot of changes to the AgentGPT since that video is made. So, the updated version will come soon and it will be filmed better.
> 
> Yes, an updated video would be much appreciated!

Hi I updated the video, it is meant for windows but for linux the process is very similar: https://youtu.be/BLIDL0E3Fxk
